tests/acceptance/dex/dex_test.go

```diff
@@ -3,134 +3,141 @@ package dex
 import (
 	"context"
 	"crypto/tls"
-	"io"
-	"io/ioutil"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
 	"net"
 	"net/http"
-	"net/url"
 	"os"
+	"os/exec"
 	"strconv"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/golang/glog"
 	. "github.com/smartystreets/goconvey/convey"
-	"golang.org/x/net/html"
 )
 
 const (
-	ingressControllerServiceURL = "istio-ingress.istio-system.svc.cluster.local"
-	domainEnvName               = "KYMA_DOMAIN"
-	isLocalEnvEnvName           = "IS_LOCAL_ENV"
-	crossClient                 = "kyma-client"
-	username                    = "admin@kyma.cx"
-	password                    = "nimda123"
+	ingressGatewayControllerServiceURLEnv = "INGRESSGATEWAY_FQDN"
+	domainEnvName                         = "KYMA_DOMAIN"
+	isLocalEnvEnvName                     = "IS_LOCAL_ENV"
+	clientId                              = "kyma-client"
+	username                              = "admin@kyma.cx"
+	password                              = "nimda123"
 )
 
 func TestSpec(t *testing.T) {
 
+	ingressGatewayControllerServiceURL, envFound := os.LookupEnv(ingressGatewayControllerServiceURLEnv)
+	if !envFound {
+		t.Fatal(ingressGatewayControllerServiceURLEnv + " env variable not set")
+	}
+
 	isLocalEnv, envFound := os.LookupEnv(isLocalEnvEnvName)
 	if !envFound {
 		t.Fatal(isLocalEnvEnvName + " env variable not set")
 	}
 
-	if testRunningInLocalEnv, err := strconv.ParseBool(isLocalEnv); err != nil {
+	testRunningInLocalEnv, err := strconv.ParseBool(isLocalEnv)
+	if err != nil {
 		t.Fatal(err)
 	} else if !testRunningInLocalEnv {
 		t.Skip()
 	}
 
-	ingressControllerAddr, err := net.LookupHost(ingressControllerServiceURL)
+	ingressGatewayControllerAddr, err := net.LookupHost(ingressGatewayControllerServiceURL)
 	if err != nil {
-		t.Fatal(err)
+
+		glog.Warningf("Unable to resolve host '%s' (if you are running this test from outside of Kyma ignore this log). Root cause: %v", ingressGatewayControllerServiceURL, err)
+
+		if minikubeIp := tryToGetMinikubeIp(); minikubeIp != "" {
+			ingressGatewayControllerAddr = []string{minikubeIp}
```

Comment:
What's the point of putting the IP in slice? Later you're just accessing the element from slice.

Reply:
The point of doing that is that ingressGatewayControllerAddr is initialized as slice a few lines above when doing net.LookupHost, so I can't change it in a runtime. 
Maybe the better approach is to save net.LookupHost in some temporary variable, define ingressGatewayControllerAddr as an uninitialized string and initialize it if err == nil or after getting minikube IP. I will use that approach in the next commit, so you can see if that approach will be better.